frontend/javascripts/oxalis/model/sagas/dataset_saga.ts

```diff
@@ -154,22 +157,27 @@ export function* watchZ1Downsampling(): Saga<void> {
 }
 
 export function* ensureSegmentIndexIsLoaded(): Saga<void> {
-  function* maybeFetchHasSegmentIndex(): Saga<void> {
-    const visibleSegmentationLayer = yield* select((state) => getVisibleSegmentationLayer(state));
-    const maybeIsSegmentIndexAvailable = yield* select((state) =>
-      getMaybeSegmentIndexAvailability(state.dataset, visibleSegmentationLayer?.name),
+  function* maybeFetchHasSegmentIndex(action: EnsureSegmentIndexIsLoadedAction): Saga<void> {
+    const { layerName } = action;
+    const dataset = yield* select((state) => state.dataset);
+    if (layerName == null) return;
+    const concernedSegmentationLayer = yield* call(getLayerByName, dataset, layerName);
```

Comment:
I'd rename it to just `segmentationLayer`. "Concerned" sounds a bit concerning :sweat_smile: